overlord/snapstate/handlers.go

```diff
@@ -753,17 +755,53 @@ func (m *SnapManager) doDownloadSnap(t *state.Task, tomb *tomb.Tomb) error {
 		if err != nil {
 			return err
 		}
+
+		// Re-compute icon download filepath and URL if the result info has the
```

Comment:
this is a very old compatibility path I'm not sure we should worry adding icon support here